source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -415,6 +434,84 @@ void ProcessorState::continueIfNecessary() {
   }
 }
 
+bool ProcessorState::handleStreamedBodyResponse(const CommonResponse& common_response) {
+  Buffer::OwnedImpl chunk_data;
+  QueuedChunkPtr chunk = dequeueStreamingChunk(chunk_data);
+  ENVOY_BUG(chunk != nullptr, "Bad streamed body callback state");
+  if (common_response.has_body_mutation()) {
+    ENVOY_LOG(debug, "Applying body response to chunk of data. Size = {}", chunk->length);
+    MutationUtils::applyBodyMutations(common_response.body_mutation(), chunk_data);
+  }
+  bool should_continue = chunk->end_stream;
+  if (chunk_data.length() > 0) {
+    ENVOY_LOG(trace, "Injecting {} bytes of data to filter stream", chunk_data.length());
+    injectDataToFilterChain(chunk_data, chunk->end_stream);
+  }
+
+  if (queueBelowLowLimit()) {
+    clearWatermark();
+  }
+  if (chunk_queue_.empty()) {
+    onFinishProcessorCall(Grpc::Status::Ok);
+  } else {
+    onFinishProcessorCall(Grpc::Status::Ok, callback_state_);
+  }
+
+  return should_continue;
+}
+
+bool ProcessorState::handleDuplexStreamedBodyResponse(const CommonResponse& common_response) {
```

Comment:
qq: Do you have test for out-of-order server response (i.e., header, body, trailer) case? It will be good to have such test, as validation of edge/error cases handing.

 Given this feature is still hide-not-implemented, i think I am fine with doing this in next PR.

Reply:
I have negative test cases for server using wrong API to send response.
However, I don't have negative test cases to test out-of-order cases, like body  response before header response, or trailer before body.  I filed this issue to track this work: https://github.com/envoyproxy/envoy/issues/37064. Let's do it in a follow up PR.